source/tests/unit/ni_fake_non_ivi_service_tests.cpp

```diff
@@ -105,6 +385,80 @@ TEST_F(NiFakeNonIviServiceTests, InitWithHandleNameAsSessionName_CloseSession_Cl
 
   EXPECT_EQ(kDriverSuccess, status);
 }
+
+TEST_F(NiFakeNonIviServiceTests, InputArraysWithNarrowIntegerTypes_U16DataGetsCoerced)
+{
+  auto status = input_arrays_with_narrow_integer_types_u16();
```

Comment:
The structure of these test methods is a little confusing.

I think you can just inline all of the helper methods into the test methods if you're not sharing them. The test methods will have access to the protected members.

If you are sharing them you can have the response as an out param. Although it's harder to pull out the EXPECT cleanly if that's doing something significant.